core/src/main/scala/kafka/log/ProducerIdMapping.scala

```diff
@@ -261,7 +261,7 @@ class ProducerIdMapping(val config: LogConfig,
         case None =>
           lastSnapOffset = 0L
           lastMapOffset = 0L
-          snapDir.mkdir()
+          Files.createDirectories(snapDir.toPath)
```

Comment:
I'm a bit unclear why we need to do this again after we do it in the constructor. cc @apurvam